framework/quota/test/org/apache/cloudstack/quota/QuotaManagerImplTest.java

```diff
@@ -0,0 +1,203 @@
+        Mockito.when(accountDao.listAll()).thenReturn(accountVOList);
+
+        UsageVO usageVO = new UsageVO();
+        usageVO.setQuotaCalculated(0);
+        List<UsageVO> usageVOList = new ArrayList<UsageVO>();
+        usageVOList.add(usageVO);
+        Pair<List<? extends UsageVO>, Integer> usageRecords = new Pair<List<? extends UsageVO>, Integer>(usageVOList, usageVOList.size());
+        Mockito.when(usageDao.getUsageRecordsPendingQuotaAggregation(Mockito.anyLong(), Mockito.anyLong())).thenReturn(usageRecords);
+
+        QuotaUsageVO quotaUsageVO = new QuotaUsageVO();
+        quotaUsageVO.setAccountId(2L);
+        List<QuotaUsageVO> quotaListForAccount = new ArrayList<>();
+        quotaListForAccount.add(quotaUsageVO);
+        Mockito.doReturn(quotaListForAccount).when(quotaManager).aggregatePendingQuotaRecordsForAccount(Mockito.eq(accountVO), Mockito.eq(usageRecords));
+        Mockito.doNothing().when(quotaManager).processQuotaBalanceForAccount(Mockito.eq(accountVO), Mockito.eq(quotaListForAccount));
+
+        assertTrue(quotaManager.calculateQuotaUsage());
+    }
+
+    @Test
+    public void testAggregatePendingQuotaRecordsForAccount() {
+        AccountVO accountVO = new AccountVO();
+        accountVO.setId(2L);
+        accountVO.setDomainId(1L);
+        accountVO.setType(Account.ACCOUNT_TYPE_NORMAL);
+
+        UsageVO usageVO = new UsageVO();
+        usageVO.setQuotaCalculated(0);
+        usageVO.setUsageType(UsageTypes.ALLOCATED_VM);
+        List<UsageVO> usageVOList = new ArrayList<UsageVO>();
+        usageVOList.add(usageVO);
+        Pair<List<? extends UsageVO>, Integer> usageRecords = new Pair<List<? extends UsageVO>, Integer>(usageVOList, usageVOList.size());
+
+        QuotaUsageVO quotaUsageVO = new QuotaUsageVO();
+        quotaUsageVO.setAccountId(2L);
+        Mockito.doReturn(quotaUsageVO).when(quotaManager).updateQuotaAllocatedVMUsage(Mockito.eq(usageVO), Mockito.any(BigDecimal.class));
+
+        assertTrue(quotaManager.aggregatePendingQuotaRecordsForAccount(accountVO, new Pair<List<? extends UsageVO>, Integer>(null, 0)).size() == 0);
+        assertTrue(quotaManager.aggregatePendingQuotaRecordsForAccount(accountVO, usageRecords).size() == 1);
+    }
+
+    @Test
+    public void testProcessQuotaBalanceForAccount() {
+        AccountVO accountVO = new AccountVO();
+        accountVO.setId(2L);
+        accountVO.setDomainId(1L);
+        accountVO.setType(Account.ACCOUNT_TYPE_NORMAL);
+
+        QuotaUsageVO quotaUsageVO = new QuotaUsageVO();
+        quotaUsageVO.setAccountId(2L);
+        quotaUsageVO.setStartDate(new Date());
+        quotaUsageVO.setEndDate(new Date());
+        List<QuotaUsageVO> quotaListForAccount = new ArrayList<>();
+        quotaListForAccount.add(quotaUsageVO);
+
+        quotaManager.processQuotaBalanceForAccount(accountVO, quotaListForAccount);
+        Mockito.verify(quotaAcc, Mockito.times(1)).persistQuotaAccount(Mockito.any(QuotaAccountVO.class));
```

Comment:
Right now there is a check if the new QuotaAccountVO object was saved in the DB. To actually check for records aggregation this code needs to be refactored.